### PR TITLE
JUnit5 migration

### PR DESCRIPTION
**Pull request by violetagg**

### CODE REVIEW

File: reactor-netty-http/src/test/java/reactor/netty/http/HttpProtocolsTests.java [hunk 1]
```diff
@@ -49,16 +50,15 @@
...
-	public HttpServer server;
-
-	@Parameter(1)
-	public HttpClient client;
+	@Retention(RetentionPolicy.RUNTIME)
+	@Target(ElementType.METHOD)
+	@ParameterizedTest(name = "{index}: {0}, {1}")
+	@MethodSource("data")
+	@interface ParameterizedHttpProtocolsTest {
+	}
```
Comment (simonbasle) [inline reactor-netty-http/src/test/java/reactor/netty/http/HttpProtocolsTests.java:60]: nice 😄

File: reactor-netty-http/src/test/java/reactor/netty/http/server/HttpServerTests.java [hunk 2]
```diff
@@ -1998,19 +2007,21 @@ private void doTestStatus(HttpResponseStatus status) {
...
-				          .port(0)
-				          .host("0.0.0.0")
-				          .route(r -> r.get("/data", (request, response) -> response.send(Mono.empty())))
-				          .wiretap(true)
-				          .bindNow();
-
-		assertThat(httpServer).isNotNull();
-
-		httpServer.disposeNow();
+		Assertions.assertTimeout(Duration.ofSeconds(10), () -> {
```
Comment (simonbasle) [inline reactor-netty-http/src/test/java/reactor/netty/http/server/HttpServerTests.java:2012]: same as before, `@Timeout(10)` annotation would do the trick

**Review by simonbasle — CHANGES_REQUESTED** [review]

**Review by aneveu — APPROVED** [review]

File: reactor-netty-http-brave/build.gradle [hunk 3]
```diff
@@ -25,6 +25,7 @@ dependencies {
 
 	testCompile "io.zipkin.brave:brave-instrumentation-http-tests:$braveVersion"
 
+	testRuntimeOnly "org.junit.vintage:junit-vintage-engine:$junitVersion"
```
[the remark this answers is not in the document]
Comment (violetagg) [reply reactor-netty-http-brave/build.gradle:28]: we always extend the brave abstract test

File: build.gradle [hunk 4]
```diff
@@ -193,7 +193,9 @@ subprojects {
 				logger.error("ERROR: Test: " + descriptor + " produced resource leak: " + event.message)
 			}
 		}
+	}
 
+	project.tasks.withType(Test).all {
```
Comment (simonbasle) [inline build.gradle:198]: sorry I should have been clearer: there's already such a block on line 163 above 😄

**Review by simonbasle — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 4 above)
Comment (violetagg) [reply build.gradle]: oops
fixed

**Review by simonbasle — APPROVED** [review]

**Pull request merged**